learn/hijack-escrow/hack-contract.md

```diff
@@ -0,0 +1,316 @@
+        messages: vec![CosmosMsg::Bank(BankMsg::Send {
+            from_address: from_human,
+            to_address: to_human,
+            amount,
+        })],
+        log,
+        data: None,
+    };
+    Ok(r)
+}
+```
+
+Note that `Env` encodes a lot of information from the blockchain, essentially providing the `Context` if you are coming from Cosmos SDK. This is validated data and can be trusted to compare any messages against. Refer to [the standard `cosmwasm` types](https://github.com/CosmWasm/cosmwasm/blob/v0.10.0/packages/std/src/types.rs#L7-L41) for references to all the available types in the environment.
+
+## Adding a New Message
+
+In this example, we will modify this contract to add some more functionality. In particular, let's make a backdoor to the contract. In the form of a `HandleMsg::Steal` variant that must be signed by some hard coded `THIEF` address and will release the entire contract balance to an address included in the message. Simple?
+
+Note that this also demonstrates the need to verify the code behind a contract rather than just rely on raw wasm. Since we have a reproducible compilation step (details below), if I show you code I claim to belong to the contract, you can compile it and compare the hash to the hash stored on the blockchain, to verify that this really is the original rust code. We will be adding tooling to automate this step and make it simpler in the coming months, but for now, this example serves to demonstrate why it is important.
+
+### Adding the Handler
+
+Open up `src/msg.rs` in your [editor of choice](./intro#setting-up-your-ide) and let's add another variant to the `HandleMsg` enum, called `Steal`. Remember, it must have a destination address:
+
+[Need a hint?](./edit-escrow-hints#handlemsg)
+
+Now, you can add the message handler. As a quick check, try running `cargo wasm` or look for the compile error in your IDE. Remember what I told you about `match`? Okay, now, add a function to process the `HandleMsg::Steal` variant. For the top level `THIEF`, you can use a placeholder address (we will set this to an address you own before deploying).
+
+[Need a hint?](./edit-escrow-hints#adding-handler)
+
+Once you are done, check that it compiles:
+
+```bash
+cargo wasm
+```
+
+### Writing a Test
+
+We have a number of tests inside of `contracts.rs` that serve as templates, so let's make use of them. You can copy the `handle_refund` test and rename it to `handle_steal`. Remember to include the `#[test]` declaration on top. Now, go in and edit it to test that the THIEF can indeed steal the funds, and no one else can. Make sure our backdoor is working properly before we try to use it.
+
+Now, try running `cargo unit-test` and see if your code works as planned. If it fails, try `RUST_BACKTRACE=1 cargo unit-test` to get a full stack trace. Now, isn't that nicer than trying to test Solidity contracts?
+
+[See solution here](./edit-escrow-hints#test-steal)
+
+### Checking Gas Usage
```

Comment:
From this point on, it is all advanced usage. 

I would pull this out and add it in another section (when writing full contract and deploying it). Otherwise too many details at the intro.